src/test/java/com/team4/testingsystem/services/TestsServiceTest.java

```diff
@@ -82,30 +83,26 @@ void getByIdFail() {
     void createWhenAssignFail() {
 
         //User doesn't exist
-        Mockito.when(usersRepository.findById(42L)).thenThrow(NoSuchElementException.class);
+        Mockito.when(usersRepository.findById(42L)).thenThrow(UserNotFoundException.class);
 
         Assertions.assertThrows(UserNotFoundException.class, () -> testsService.createForUser(42L));
     }
 
     @org.junit.jupiter.api.Test
     void createSuccess() {
 
-
         Mockito.when(usersRepository.findById(1L)).thenReturn(Optional.of(user));
 
-        MockedStatic<Test> testMockedStatic = Mockito.mockStatic(Test.class);
-        testMockedStatic.when(Test::newBuilder).thenReturn(builder);
-
-        Mockito.when(builder.setUser(any())).thenReturn(builder);
-        Mockito.when(builder.setCreatedAt(any())).thenReturn(builder);
-        Mockito.when(builder.setStatus(any())).thenReturn(builder);
-        Mockito.when(builder.build()).thenReturn(test);
-
-        Mockito.when(test.getId()).thenReturn(1L);
-
-
-        Assertions.assertEquals(1L, testsService.createForUser(1L));
-
+        try (MockedConstruction<Test.Builder> builderMockedConstruction = Mockito.mockConstruction(Test.Builder.class,
+                (mock, context) -> {
+                    Mockito.when(mock.user(any())).thenReturn(mock);
+                    Mockito.when(mock.createdAt(any())).thenReturn(mock);
+                    Mockito.when(mock.status(any())).thenReturn(mock);
+                    Mockito.when(mock.build()).thenReturn(test);
+                })) {
+            Mockito.when(test.getId()).thenReturn(1L);
+            Assertions.assertEquals(1L, testsService.createForUser(1L));
+        }
```

Comment:
Let's leave static method for creating builder in `Test` class:
```
public static Builder builder() {
    return new Builder();
}
````

Then, you can use builder with `Test.builder()` instead of `new Test.Builder()`.
Moreover, this large constructor mocking can be replaced with mocking static method `builder()`, like it was before.
P.S. Mocking static method should be done with try-block :)